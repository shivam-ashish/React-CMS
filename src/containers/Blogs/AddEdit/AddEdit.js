/* eslint-disable no-alert */
import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
// import Edit from './Edit';
import firebase from '../../../config/fire';
import classes from './AddEdit.module.scss';
import Button from '../../../commonComponents/Button/Button';
import BtnClass from '../../../commonComponents/Button/Button.module.scss';

class AddNewPost extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      body: '',
      timeStamp: '',
    };
  }

  componentDidMount() {
    console.log('inside addEdit compoDidMount', this.props);
    const { editObject } = this.props;

    let updateState = {};

    if (editObject) {
      updateState = { ...editObject };
    }

    this.setState({
      ...updateState,
      timeStamp: `${new Date().getTime()}`,
    });
  }

  putData = () => {
    const { title, body, timeStamp } = this.state;
    const { props } = this;
    const { uid } = props.user;
    const database = firebase.database();
    const ref = database.ref('blogs');
    const data = {
      title,
      body,
      submittedBy: uid,
      submittedOn: timeStamp,
      updatedBy: uid,
      updatedOn: timeStamp,
    };
    if (title && body != null) {
      ref.push(data);
    } else {
      alert('Plzzz Enter Some Value!!!');
    }
  }

  editData = () => {
    const { props } = this;
    const { key } = props.match.params;
    const { uid } = props.user;
    const { title, body, timeStamp } = this.state;
    firebase.database().ref(`blogs/${key}`).update({
      title,
      body,
      updatedBy: uid,
      updatedOn: timeStamp,
    });
  }


  handleChange = ({ target: { value, name } }) => {
    this.setState({ [name]: value });
  }

  addOrEdit = (type) => {
    switch (type) {
      case 'add': return (
        <Button
          type="Add Your Post"
          className={BtnClass.addEdit}
          click={this.putData}
        >
          {'Add Your Post'}
        </Button>
      );
      case 'edit': return (
        <Button
          type="Edit Your Post"
          className={BtnClass.addEdit}
          click={this.editData}
        >
          {'Edit Your Post'}
        </Button>
      );
      default: return null;
    }
  }

  render() {
    const { props } = this;
    const { type } = props.match.params;
    const { title, body } = this.state;

    return (
      <div className={classes.post}>
        <h1 className={classes.heading}>
          {type.toUpperCase()}
          {' '}
          YOUR POST
        </h1>
        <div className={classes.container}>
          <label className={classes.labelField} htmlFor="title">Title</label>
          <br />
          <input
            className={classes.inputField}
            onChange={this.handleChange}
            type="text"
            value={title}
            name="title"
          />
          <br />
          <label className={classes.labelField} htmlFor="body">
            Body
          </label>
          <br />
          <input
            className={classes.inputField}
            onChange={this.handleChange}
            type="text"
            value={body}
            name="body"
          />
          <br />
          <Link to="/home/blogs">
            {this.addOrEdit(type)}
          </Link>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  user: state.reducer.user,
});

export default withRouter(connect(mapStateToProps)(AddNewPost));
