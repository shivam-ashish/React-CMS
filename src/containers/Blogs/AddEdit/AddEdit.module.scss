@import "../../Sass/main.scss";

.post{
    .heading{
    margin: 0px;
    padding: 0px;
    text-align: center;
    font-size: $blogs-news-heading-font;
    color: $blogs;
    font-weight: 600;
    }

    .inputField{
    display: inline-block;
    border: none;
    border-bottom: 1px solid $grey;
    background: transparent;
    outline: none;
    height: 30px;
    color: $black;
    font-size: $input-type-font;
    margin-bottom: 20px;
    display: block;
    margin-bottom: 20px;
    width: 90%;
    }

    .labelField{
    font-size: $label-font;
    margin: 0;
    padding: 0;
    font-weight: 500;
    color: $black;
    }
}
.container{
    width: 50%;
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 350px;
    height: auto;
    box-sizing: border-box;
    background: $white;
    padding: 20px 40px;
    box-shadow: 10px 6px 9px 1px $grey;
}
