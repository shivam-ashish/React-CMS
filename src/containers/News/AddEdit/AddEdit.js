import React, { Component } from 'react';
import Fire from '../../../config/fire';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import classes from './AddEdit.module.css';

class AddNewPost extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      body: '',
    };
  }

  componentDidMount() {
    this.setState({
      timeStamp: `${new Date().getTime()}`,
    });
  }

  putData = () => {
    const { title, body, timeStamp } = this.state;
    const { uid } = this.props.user;
    const database = Fire.database();
    const ref = database.ref('news');
    const data = {
      title,
      body,
      submittedBy: uid,
      submittedOn: timeStamp,
      updatedBy: uid,
      updatedOn: timeStamp,
    };
    ref.push(data);
  }

  editData = () => {
    const { key } = this.props.match.params;
    const { uid } = this.props.user;
    const { title, body, timeStamp } = this.state;
    Fire.database().ref(`news/${key}`).update({
      title,
      body,
      updatedBy: uid,
      updatedOn: timeStamp,
    });
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    const { type } = this.props.match.params;
    const { title, body } = this.state;
    return (
      <div className={classes.post}>
        <h1>
          { type }
        YOUR NEWS
        </h1>
        <div className={classes.container}>
          <label htmlFor="title">Title</label>
          <br />
          <input
            value={title}
            onChange={this.handleChange}
            type="text"
            placeholder="Enter Title"
            name="title"
          />
          <br />
          <label htmlFor="body">
            Body
          </label>
          <br />
          <input
            value={body}
            onChange={this.handleChange}
            type="text"
            placeholder="Enter your Post"
            name="body"
            className={classes.body}
          />
          <br />
          <Link to="/home/news">


            {(() => {
              switch (type) {
                case 'add': return (
                  <button
                    type="button"
                    onClick={this.putData}
                  >
                    { type }
                     NEWS
                  </button>
                );
                case 'edit': return (
                  <button
                    type="button"
                    onClick={this.editData}
                  >
                    { type }
                     NEWS
                  </button>
                );
                default: return null;
              }
            })()}
          </Link>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user,
});

export default withRouter(connect(mapStateToProps)(AddNewPost));
