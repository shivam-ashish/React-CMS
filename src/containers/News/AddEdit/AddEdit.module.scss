@import "../../Sass/main.scss";

.post{ 
    h1{
    margin: 0px;
    padding: 0px;
    text-align: center;
    font-size: $blogs-news-heading-font;
    color: $news;
    font-weight: 600;
    }
    input[type="text"]{
    display: inline-block;
    border: none;
    border-bottom: 1px solid $grey;
    background: transparent;
    outline: none;
    height: 30px;
    color: $label;
    font-size: $input-type-font;
    margin-bottom: 20px;
    }
    input {
    display: block;
    margin-bottom: 20px;
    width: 90%;
    }
    label{
    font-size: $label-font;
    margin: 0;
    padding: 0;
    font-weight: 500;
    color: $black;
    }
    button{
    border: none;
    outline: none;
    height: 40px;
    color: $buttonText;
    font-size: $button-font;
    background: $addEdit;
    cursor: pointer;
    width: 100%;
    margin-top: 20px;
    border-radius: 10px;
    &:hover{
    background: $addEditHover;
    }
}
}
.container{
    width: 50%;
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 350px;
    height: auto;
    box-sizing: border-box;
    background: $white;
    padding: 20px 40px;
    box-shadow: 10px 6px 9px 1px $grey;
}