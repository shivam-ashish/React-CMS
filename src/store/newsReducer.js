const initialState = {
  news: null,
};

const newsReducer = (state = initialState, { payload }) => {
  return {
    ...state,
    news: payload,
  };
};

export default newsReducer;
