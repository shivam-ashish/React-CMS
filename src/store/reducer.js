
const initialState ={
    isLoggedIn: false,
    user: 'user1',
}

const reducer = ( state = initialState, action ) => {
    const newState = { ...state };
    switch (action.type){
    case "LOGIN_STATE":
    return{
        ...state,
        isLoggedIn: Boolean,
        history: state.history.concat({
            isLoggedIn: Boolean
        })
    };
    case "UPDATE_USER":
    return{
        ...state,
        
        history: state.history.concat({
            
        })
    };
    default:
    return newState;
}
};

export default reducer;