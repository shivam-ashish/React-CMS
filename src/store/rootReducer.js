import { createStore, combineReducers } from 'redux';
import reducer from './reducers/reducer';
import newsReducer from './reducers/newsReducer';
import blogReducer from './reducers/blogReducer';

const rootReducer = combineReducers({
  reducer,
  newsReducer,
  blogReducer,
});

const store = createStore(rootReducer);

export default store;
