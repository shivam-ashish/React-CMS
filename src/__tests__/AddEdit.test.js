import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import AddEdit from '../containers/Blogs/AddEdit/AddEdit';
import store from '../store/rootReducer';

let wrapped;

beforeEach(() => {
  wrapped = mount(<Provider store={store}><Router><AddEdit props="xyz" /></Router></Provider>);
});
afterEach(() => {
  wrapped.unmount();
});

it('has a AddEdit component structure', () => {
  expect(wrapped.find('div').length).toEqual(2);
  expect(wrapped.find('h1').length).toEqual(1);
  expect(wrapped.find('input').length).toEqual(2);
});

it('has text fields where users can type in', () => {
  wrapped.find('#title').simulate('change', {
    target: { value: 'new title', name: 'title' },
  });
  wrapped.find('#body').simulate('change', {
    target: { value: 'new body', name: 'body' },
  });
  wrapped.update();
  expect(wrapped.find('#title').prop('value')).toEqual('new title');
  expect(wrapped.find('#body').prop('value')).toEqual('new body');
});
