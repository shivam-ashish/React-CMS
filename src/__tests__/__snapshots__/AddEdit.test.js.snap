// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<div
  className="post"
>
  <h1
    className="heading"
  >
     
    YOUR POST
  </h1>
  <div
    className="container"
  >
    <label
      className="labelField"
      htmlFor="title"
    >
      Title
    </label>
    <br />
    <input
      className="inputField"
      id="title"
      name="title"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <label
      className="labelField"
      htmlFor="body"
    >
      Body
    </label>
    <br />
    <input
      className="inputField"
      id="body"
      name="body"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <a
      href="/home/blogs"
      onClick={[Function]}
    />
  </div>
</div>
`;
